examples/mauna-loa/script.jl

```diff
@@ -0,0 +1,302 @@
+#     function as part of JuliaGaussianProcesses -- for now, we just define it
+#     inline.
+#
+# The L-BFGS parameters were chosen because they seem to work well empirically.
+# You could also try with the defaults.
+
+default_optimizer = LBFGS(;
+    alphaguess=Optim.LineSearches.InitialStatic(; scaled=true),
+    linesearch=Optim.LineSearches.BackTracking(),
+)
+
+function optimize_loss(loss, θ_init; optimizer=default_optimizer, maxiter=1_000)
+    options = Optim.Options(; iterations=maxiter, show_trace=true)
+
+    θ_flat_init, unflatten = ParameterHandling.value_flatten(θ_init)
+    loss_packed = loss ∘ unflatten
+
+    ## https://julianlsolvers.github.io/Optim.jl/stable/#user/tipsandtricks/#avoid-repeating-computations
+    function fg!(F, G, x)
+        if F !== nothing && G !== nothing
+            val, grad = Zygote.withgradient(loss_packed, x)
+            G .= only(grad)
+            return val
+        elseif G !== nothing
+            grad = Zygote.gradient(loss_packed, x)
+            G .= only(grad)
+            return nothing
+        elseif F !== nothing
+            return loss_packed(x)
+        end
+    end
+
+    result = optimize(Optim.only_fg!(fg!), θ_flat_init, optimizer, options; inplace=false)
+
+    return unflatten(result.minimizer), result
+end
+#md nothing #hide
+
+# We now run the optimization:
+
+θ_opt, result = optimize_loss(loss, θ_init)
+result
+
+# The final value of the log marginal likelihood is:
+
+-result.minimum
+
+# !!! warning
+#     To avoid bad local optima, we could (and should) have carried out several
+#     random restarts with different initial values for the hyperparameters,
+#     and then picked the result with the highest marginal likelihood. We omit
+#     this for simplicity. For more details on how to fit GPs in practice,
+#     check out [A Practical Guide to Gaussian
+#     Processes](https://tinyurl.com/guide2gp).
+#
+# Let's construct the posterior GP with the optimized hyperparameters:
+
+fpost_opt = build_posterior_gp(ParameterHandling.value(θ_opt))
+#md nothing #hide
+
+# This is the kernel with the point-estimated hyperparameters:
+
+fpost_opt.prior.kernel
+
+# Let's print the optimized values of the hyperparameters in a more helpful format:
+
+# !!! note "Work-in-progress"
+#     This is another utility function we would eventually like to move out of this notebook:
+
+using Printf
+
+function show_params(nt::NamedTuple, pre=0)
+    res = ""
+    for (s, v) in pairs(nt)
+        if typeof(v) <: NamedTuple
+            res *= join(fill(" ", pre)) * "$(s):\n" * show_params(v, pre + 4)
+        else
+            res *= join(fill(" ", pre)) * "$s = $(@sprintf("%.3f", v))\n"
+        end
+    end
+    return res
+end
+
+print(show_params(ParameterHandling.value(θ_opt)))
+
+# And, finally, we can visualize our optimized posterior GP:
+
+let
+    plotdata()
+    label = "optimized posterior f(⋅)"
+    plot_gp!(fpost_opt; label)
+end
```

Comment:
Same here, this is so unidiomatic that I really don't think it should be used in a tutorial (additionally, as mentioned above `; label)` is only supported in new versions). Please just use
```suggestion
plotdata()
plot_gp!(fpost_opt; label="optimized posterior f(⋅)")
```